src/controllers/ping.controller.ts

```diff
@@ -0,0 +1,283 @@
+import Ping from '../models/Ping.model';
+import { Request, Response } from 'express';
+import { KeyValidationType, verifyKeys } from '../utils/validation.utils';
+import User from '../models/User.model';
+import { addPingExpireJob } from '../queue/jobs';
+import { sendNotifications } from '../utils/notification.utils';
+import { NotificationType } from '../interfaces/Notification.interface';
+import { PingStatus } from '../interfaces/Ping.interface';
+import mongoose from 'mongoose';
+import { nightlightQueue } from '../queue/setup/queue.setup';
+
+/**
+ * Sends a ping from the sender to the recipient.
+ * Adds a queue job to change the ping status after the expiration date.
+ * Sends a notification to the recipient about the ping.
+ *
+ * @param {Request} req - Express request object containing the ping object data
+ * @param {Response} res - Express response object containing the ping object data after it has been saved to the database
+ * @returns {Promise<Ping>} - Resolved promise containing the ping object data after it has been saved to the database
+ */
+export const sendPing = async (req: Request, res: Response) => {
+  // Get the ping object from the request body
+  const pingData = req.body;
+
+  const sentDateTime = new Date().toUTCString();
+
+  // Verify that the ping object has all the necessary keys
+  const validationError = verifyKeys(pingData, KeyValidationType.PING);
+  if (validationError !== '') {
+    return res.status(400).send({ message: validationError });
+  }
+
+  try {
+    // Check if the sender id is valid
+    const ping = { ...pingData, sentDateTime, status: PingStatus.SENT };
+
+    // Create a new ping from the request body
+    const newPing = new Ping(ping);
+
+    // Save the ping to the database
+    const savedPing = await newPing.save();
+
+    // Check if the ping was saved
+    if (savedPing === null) {
+      return res.status(500).send({ message: 'Ping could not be saved' });
+    }
+
+    // Calculate the delay for the ping expiration
+    const delay =
+      new Date().getTime() - new Date(ping.expirationDateTime).getTime();
+
+    // Add the ping expiration to the queue
+    const job = await addPingExpireJob(savedPing._id.toString(), delay);
+
+    // Check if the job was added to the queue
+    if (job === null || job?.id === undefined) {
+      return res
```

Comment:
job failed to add but we still end up having a `Ping` inserted in our database (line 41). should that be the behavior?